brigade.js

```diff
@@ -28,166 +17,13 @@ function build(e, project) {
-        // for (re-)triggering the default Checks suite
-        case "/brig run":
-            return runSuite(e, p);
-        default:
-            console.log(`No applicable action found for comment: ${comment}`);
-    }
-}
-
-// Helpers
-// TODO: use/consume corresponding utils in this project for this (and other?) logic
-
-// A GitHub Check Suite notification
-class Notification {
-    constructor(name, e, p) {
-        this.proj = p;
-        this.payload = e.payload;
-        this.name = name;
-        this.externalID = e.buildID;
-        this.detailsURL = `https://brigadecore.github.io/kashti/builds/${e.buildID}`;
-        this.title = "running check";
-        this.text = "";
-        this.summary = "";
-
-        // count allows us to send the notification multiple times, with a distinct pod name
-        // each time.
-        this.count = 0;
-
-        // One of: "success", "failure", "neutral", "cancelled", or "timed_out".
-        this.conclusion = "neutral";
-    }
-
-    // Send a new notification, and return a Promise<result>.
-    run() {
-        this.count++
-        var j = new Job(`${this.name}-${this.count}`, "brigadecore/brigade-github-check-run:latest");
-        j.imageForcePull = true;
-        j.env = {
-            CHECK_CONCLUSION: this.conclusion,
-            CHECK_NAME: this.name,
-            CHECK_TITLE: this.title,
-            CHECK_PAYLOAD: this.payload,
-            CHECK_SUMMARY: this.summary,
-            CHECK_TEXT: this.text,
-            CHECK_DETAILS_URL: this.detailsURL,
-            CHECK_EXTERNAL_ID: this.externalID
-        }
-        return j.run();
-    }
-}
-
-// Helper to wrap a job execution between two notifications.
-async function notificationWrap(job, note, conclusion) {
-    if (conclusion == null) {
-        conclusion = "success"
-    }
-    await note.run();
-    try {
-        let res = await job.run()
-        const logs = await job.logs();
-
-        note.conclusion = conclusion;
-        note.summary = `Task "${job.name}" passed`;
-        note.text = "```" + res.toString() + "```\nTest Complete";
-        return await note.run();
-    } catch (e) {
-        const logs = await job.logs();
-        note.conclusion = "failure";
-        note.summary = `Task "${job.name}" failed for ${e.buildID}`;
-        note.text = "```" + logs + "```\nFailed with error: " + e.toString();
-        try {
-            return await note.run();
-        } catch (e2) {
-            console.error("failed to send notification: " + e2.toString());
-            console.error("original error: " + e.toString());
-            return e2;
-        }
-    }
-}
+events.on("exec", runSuite);
+events.on("check_suite:requested", runSuite);
+events.on("check_suite:rerequested", runSuite);
+events.on("issue_comment:created", (e, p) => Check.handleIssueComment(e, p, runSuite));
+events.on("issue_comment:edied", (e, p) => Check.handleIssueComment(e, p, runSuite));
```

Comment:
typo on the action here: `s/edied/edited`

Reply:
We've lost a handler for `events.on("check_run:rerequested", runCheck);`... file an issue to add back in if not in this same PR?

Reply:
Opened #21 to track this - as we only have one check run at the moment, I don't see this as a priority, but feel free to disagree, and I'll tackle it next.